File: 284/2A.rb
n, x = gets.split(' ').map(&:to_i)
pre = 0
ans = 0
n.times do
    l, r = gets.split(' ').map(&:to_i)
    ans += (l - pre - 1) % x + r - l + 1
    pre = r
end
print ans, "\n";

File: 325/2A.pl
$n = <>;
@a = split ' ', <>;
$ans = $k = 0;
foreach $_ (@a) {
    if ($_ == '1') {
        if ($ans > 0 && $k == 1) {
            $ans ++;
        }
        $ans ++;
        $k = 0;
    } else {
        $k ++;
    }
}
print $ans, "\n";
